src/AddTaskDialog.jsx

```diff
@@ -0,0 +1,126 @@
+import Select from '@mui/material/Select';
+import MenuItem from '@mui/material/MenuItem';
+import Box from '@mui/material/Box';
+import Rating from '@mui/material/Rating';
+import { v4 as uuidv4 } from 'uuid';
+
+
+const AddTaskDialog = ({ open, handleClose, onSave, editedTask }) => {
+
+    const [formData, setFormData] = useState({
+        id: null,
+        name: '',
+        subject: '',
+        priority: 0,
+        isChecked: false,
+    });
+
+    const handleChange = (e) => {
+        const { name, value } = e.target;
+        if (name === 'priority'){
+            setFormData({
+                ...formData,
+                priority: parseInt(value)
+            })
+        }
+        else{
+            setFormData({
+                ...formData,
+                [name]: value,
+            });
+        }
+    };
+
+    const handleSave = () => {
+        onSave(formData);
+        handleClose();
+    };
+
+    useEffect(() => {
+        if (editedTask) {
+            setFormData({
+                id: editedTask.id,
+                name: editedTask.name,
+                subject: editedTask.subject,
+                priority: editedTask.priority,
+                isChecked: editedTask.isChecked,
+            });
+        } else {
+            setFormData({
+                id: uuidv4(), 
+                name: '',
+                subject: '',
+                priority: 0,
+                isChecked: false,
+            });
+        }
+    }, [editedTask]);
+
+    return (
+        <>
+            <Dialog className='dialog' open={open} onClose={handleClose} fullWidth >
+                <DialogTitle className='dialog-title'>{editedTask instanceof Event ? 'Add New Task' : 'Edit Task'}</DialogTitle >
+                <DialogContent className='dialog-content'>
+                    <div className='taskInput'>
+                        <TextField
+                            autoFocus
+                            margin="dense"
+                            id="taskName"
+                            label="Task Name"
+                            variant="standard"
+                            type="text"
+                            name='name'
+                            value={formData.name || ''}
+                            onChange={handleChange}
+                        />
+                    </div>
+                    <div className='subjectInput'>
+                        <Select
+                            labelId="demo-simple-select-label"
+                            id="demo-simple-select-standard"
+                            value={formData.subject || ''}
+                            onChange={handleChange}
+                            name='subject'
+                        >
+                            <MenuItem value="" disabled>
+                                Select Task Subject
+                            </MenuItem>
+                            <MenuItem value="Personal">Personal</MenuItem>
```

Comment:
define an array that contains all of your subject options and than give it to the select